test/integration/hotrestart_handoff_test.py

```diff
@@ -110,48 +117,102 @@ async def slow_response(self, request):
         return response
 
 
-async def _http_request(url) -> AsyncIterator[str]:
-    # Separate session per request is against aiohttp idioms, but is
-    # intentional here because the point of the test is verifying
-    # where connections go - reusing a connection would do the wrong thing.
-    async with ClientSession() as session:
-        async with session.get(url) as response:
-            async for line in response.content:
-                yield line
+class LineGenerator:
+
+    @cached_property
+    def _queue(self) -> asyncio.Queue[str]:
+        return asyncio.Queue()
+
+    @cached_property
+    def _task(self):
+        return asyncio.create_task(self.generator())
+
+    @abc.abstractmethod
+    async def generator(self) -> None:
+        pass
```

Comment:
```suggestion
        raise NotImplementedError
```

if its abstract but should always be overridden then rather make that explicit

its kinda annoying that there is not an `interfacemethod` in `std` - `abstracts` (lib in toolshed does)